### DISCUSSION TITLE
How can I define and change a theme at runtime with CDN version.

### DISCUSSION THREAD

**Discussion by Mashiane** [saadeghi/daisyui#1317] [Q&A] [answered]
Hi

Well this is not ideal but here am I.

The colors section bears reference: https://daisyui.com/docs/colors/

As an example bg-primary is defined as hsl(var(--p))

If I wanted to change the --p variable to some value at runtime to apply across my app, how can I do it.

I am using the CDN version.

Thanks in advance.

**Comment by saadeghi** [comment, marked as answer]
Hi,
Please take a look at the built-in themes here: https://unpkg.com/browse/daisyui@2.38.0/dist/themes.css  
As you can see, every color has 3 values of HSL. If you have the HSL value of your color, you can overwrite your own value to any CSS variable.

For example, for the default theme, `--p` value is `259 94% 51%;` which is the purple color.

If your colors on the runtime are not HSL (it's hex, RGB, etc.) you can convert it to HSL using a color library like this: https://www.npmjs.com/package/color
